#! /bin/sh /usr/share/dpatch/dpatch-run
## 940_hppa_disable_test_propag_signal.dpatch by Lucas Nussbaum <lucas@lucas-nussbaum.net>
##
## All lines beginning with `## DP:' are a description of the patch.
## DP: Due to problems with the threads implementation on hppa, this test
## DP: blocks. We disable it on this arch only.
## DP: Closes: #514695.

@DPATCH@
Index: b/test/ruby/test_beginendblock.rb
===================================================================
--- a/test/ruby/test_beginendblock.rb	2010-07-20 21:45:45.000000000 +0900
+++ b/test/ruby/test_beginendblock.rb	2010-07-20 21:46:16.000000000 +0900
@@ -1,4 +1,5 @@
 require 'test/unit'
+require 'rbconfig'
 require 'tempfile'
 require_relative 'envutil'
 
@@ -91,6 +92,7 @@
     assert_nil $?.termsig
   end
 
+  if Config::CONFIG['arch'] != 'hppa-linux'
   def test_should_propagate_signaled
     ruby = EnvUtil.rubybin
     out = IO.popen(
@@ -104,4 +106,5 @@
     assert_nil $?.exitstatus
     assert_equal Signal.list["INT"], $?.termsig
   end
+  end
 end
